Address issue 3 of the audit report

borgbob opened the pull request:
@Callum-A / @ezekiiel  I was thinking of calling `add_attribute("key_removed", key_removed)` in the response to indicate if the key was actually removed, but it was not clear if it would be useful information

Close #325

borgbob commented on contracts/cw-core/src/contract.rs:
```diff
@@ -379,7 +379,10 @@ pub fn execute_remove_item(
     }
 
-    ITEMS.remove(deps.storage, key.clone());
+    if ITEMS.has(deps.storage, key.clone()) {
```
I'm new to Rust, so please advise if there's a better way to do this. All this cloning is really strange just to satisfy the compiler, it seems that both `ITEMS.has` and `ITEMS.remove` only need read access to `key`, so why can't we pass it by reference?

ezekiiel replied:
I think unfortunately these clones are needed. The reason we need to clone is because the remove function takes an item of type `K` where `K` is the key being used in the map. If it took `&K` we could pass by reference, but the API tells us that the function wants to take ownership of the key. Under the hood I believe it does this because it serializes the input key into a vec.

borgbob commented:
@ezekiiel Can you review again?

I think the existing `add_remove_get` test was broken since it was trying to remove a missing key. I've fixed the test and added a new test to verify the error being returned.

ezekiiel reviewed (approved):
LG2M. Thanks!
